### ISSUE TITLE
Add `**open_options` to FileWatcher

### ISSUE THREAD

**Issue by foklepoint** [reddit/baseplate#243]
Currently, a FileWatcher can't be forced to `open()` a file with a specific encoding type. This is necessary when you don't want to fallback to the platform encoding and instead force a file to be opened as a specific encoding type (i.e: utf-8) 

A proposed solution is to add `**open_options` kwargs to [`FileWatcher.__init__`](https://github.com/reddit/baseplate/blob/8f6e05c88d00ee1e467951a577e22c25aca5ec4a/baseplate/file_watcher.py#L70)

This would also allow the FileWatcher to modify the open operation in the following ways:

```
mode='r'
buffering=-1
encoding=None
errors=None
newline=None
closefd=True
opener=None
```

**Comment by pacejackson** [comment]
Note, this is really only applicable to Python 3.  Python 3 has [several options](https://docs.python.org/3/library/functions.html#open) while Python 2 [does not](https://docs.python.org/2.7/library/functions.html#open)

**Comment by pacejackson** [comment]
Addressed in #250
